mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java

```diff
@@ -168,4 +168,20 @@ private DB extractDatabase(MongoClient mongo, MongoDBConfiguration config) {
 			}
 		}
 	}
+
+	protected MongoDBConfiguration getConfig() {
+		return config;
+	}
+
+	protected MongoClient getMongoClient() {
+		return mongo;
+	}
+
+	protected void setMongoClient(MongoClient mongoClient) {
```

Comment:
I was thinking about using Template pattern (or decorator), but @emmanuelbernard thinks it would be better to not relay on extends and use aggregation. If you prefer I use Decorator pattern, no problem.